builder/source/simple.go

```diff
@@ -112,50 +113,52 @@ func (s *SimpleSource) IsFetched() bool {
 	return PathExists(s.GetPath(s.validator))
 }
 
-// download utilises CURL to do all downloads
+// download downloads simple files using go grab
 func (s *SimpleSource) download(destination string) error {
-	hnd := curl.EasyInit()
-	defer hnd.Cleanup()
 
-	hnd.Setopt(curl.OPT_URL, s.URI)
-	hnd.Setopt(curl.OPT_FOLLOWLOCATION, 1)
+	file, err := os.Create(destination)
+	if err != nil {
+		return err
+	}
+	defer file.Close()
 
-	out, err := os.Create(destination)
+	client := grab.NewClient()
+	req, err := grab.NewRequest(destination, s.URI)
 	if err != nil {
 		return err
 	}
 
-	pbar := pb.New64(0)
-	pbar.Set(pb.Bytes, true)
-	pbar.Set("prefix", filepath.Base(destination))
-	pbar.SetMaxWidth(80)
+	resp := client.Do(req)
 
-	writer := func(data []byte, udata interface{}) bool {
-		if _, err := out.Write(data); err != nil {
-			return false
+	// Setup our progress bar
+	var pbar *pb.ProgressBar
+	pbar = pb.Start64(resp.Size())
+	pbar.Set(pb.Bytes, true)
+	// Default template except that the 'per second' suffix is changed to '/s' instead of 'p/s'
+	progressBarTemplate := `{{with string . "prefix"}}{{.}} {{end}}{{counters . }} {{bar . }} {{percent . }} {{speed . "%s/s" "??/s"}}{{with string . "suffix"}} {{.}}{{end}}`
+	pbar.SetTemplateString(progressBarTemplate)
+	defer pbar.Finish()
+
+	// Timer to integrate into pbar
+	t := time.NewTicker(100 * time.Millisecond)
+	defer t.Stop()
+
+Loop:
+	for {
+		select {
+		case <-t.C:
+			pbar.SetCurrent(resp.BytesComplete())
+		case <-resp.Done:
+			break Loop
```

Comment:
I have an irrational hatred for labels, so I would just end the function right here instead of using one:

```suggestion
			if err := resp.Err(); err != nil {
				log.Errorf("Error downloading %s: %v\n", s.URI, err)
				return err
			}

			return nil
```